### DISCUSSION TITLE
Stream Generator variable with VS1053

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [pschatzmann/arduino-audio-tools#356] [Q&A] [answered]
Hello Pshatzmann,i'm having fun with your VS1053 library and the generator,but only one frequency is to me frustrating,i tried the generator sketch with a global variable
 coming from and ADC input where a potentiometer is wired,the skech is:

//----------------------------------------------------------------

/*
 *  SCK  SPI Serial Clock ->18
    MISO  SPI Master Input ← Slave Output ->19
    MOSI  SPI Master Output → Slave Input ->23
   (X)CS   SPI Chip Select/ Configured in Sketch ->5
   (X)DCS  Data Chip Select: Configured in Sketch ->16
   DREQ  Data Request: Configured in Sketch ->4
   XRST  Hardware Reset Pin: low=reset ->15
 * 
 */
#include "AudioTools.h"
#include "AudioLibs/VS1053Stream.h"

uint16_t sample_rate=44100;
uint8_t channels = 2;                                      
uint8_t bits_per_sample = 16; 

const int potPin = 34;//potentiometer settings
int potValue; 

SineWaveGenerator<int16_t> sineWave(potValue); //variable generator   
            
GeneratedSoundStream<int16_t> sound(sineWave);             
VS1053Stream out;    // VS1053 output
StreamCopy copier(out, sound); // copy sound to decoder

void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning); 

  // Setup sine wave
  sineWave.begin(channels, sample_rate, N_A4);

  // setup output
  auto cfg = out.defaultConfig();
  cfg.sample_rate = sample_rate;
  cfg.channels = channels;
  cfg.bits_per_sample = bits_per_sample;
  // Use your custom pins or define in AudioCodnfig.h
  //cfg.cs_pin = VS1053_CS; 
  //cfg.dcs_pin = VS1053_DCS;
  //cfg.dreq_pin = VS1053_DREQ;
  //cfg.reset_pin = VS1053_RESET;
  out.begin(cfg);

}

void loop(){
  copier.copy();

  potValue = analogRead(potPin); //getting values from the pot
  Serial.println(potValue);
}

//---------------------------------------------------------------
Nothing is changed and i can't hear anything.
I know it's a candid modification of the sketch,but is it possible to vary 
the frequency?

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you have a look at the the synthesizer examples in the AudioKit directory: e.g. https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-synthbasic1-audiokit/streams-synthbasic1-audiokit.ino

**Reply by TangerineDreamer** [reply]
Thank you for this very fast reply :)
i guess i have to play with this function:

void actionKeyOn(bool active, int pin, void* ptr){
  int freq = *((int*)ptr);
  sine.setFrequency(freq);
  in.begin();
}

I'll have a try with :)
